packages/alfa-util/test/option.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { test } from "@siteimprove/alfa-test";
+import { none, option, some } from "../src/option";
+
+test("Some only runs callback if parameter is something", t => {
+  t.equal(some(4, (num: number) => "callback"), "callback");
+  t.equal(some(null, (num: number) => "callback"), null);
+});
+
+test("None only runs callback if parameter is not something", t => {
+  t.equal(none(4, () => "callback"), 4);
+  t.equal(none(null, () => "callback"), "callback");
+});
+
+test("Options runs some if parameter is something, else none", t => {
+  t.equal(
+    option(4, (num: number) => "something", () => "nothing"),
+    "something"
+  );
+  t.equal(
+    option(null, (num: number) => "something", () => "nothing"),
+    "nothing"
+  );
+});
```

Comment:
You can get around having to annotate everything by defining these at the top:

```ts
let something: Option<number> = 4;
let nothing: Option<number> = null;
```

Reply:
I think that is a great improvement. Thank you.